desc/objectives/_free_boundary.py

```diff
@@ -756,7 +756,11 @@ def compute(self, eq_params, *field_params, constants=None):
 
         g = eval_data["|e_theta x e_zeta|"]
         Bn_err = Bn * g
-        Bsq_err = (bsq_in + eval_data["p"] * (2 * mu_0) - bsq_out) * g
+        Bsq_err = jnp.where(
```

Comment:
Does this also work for reverse mode? Usually in that case you need a double where

Reply:
It does not work for reverse mode. I don't think it can be fixed with a double where because computing the pressure itself is the thing that is undefined, not some function evaluated at a pressure of 0. So we can't check "where p=0" without causing NaNs. 

My old method with the input argument flag would work with reverse mode, but I still like this new method better because it avoids any work for the user. It is unlikely that anyone would need/want to use reverse mode with this objective anyways. I could add a check to make sure `deriv_mode="fwd"`, at least if the equilibrium has kinetic profiles?